sympy/solvers/polysys.py

```diff
@@ -429,3 +439,389 @@ def solve_triangulated(polys, *gens, **args):
 
         solutions = _solutions
     return sorted((s for s, _ in solutions), key=default_sort_key)
+
+
+def factor_poly_system(eqs, *gens, **kwargs):
+    """
+    Given a system of polynomial equations, returns a factored form and solvability conditions.
+
+    Parameters
+    ==========
+
+    eqs : list
+        List of expressions to be factored. Each expression is assumed to be equal to zero.
+    *gens : Symbol or sequence of Symbols, optional
+        Generator(s) of the polynomial ring. If not provided, all free symbols will be used.
+    **kwargs : dict, optional
+        Optional keyword arguments passed to polynomial manipulation functions.
+
+    Returns
+    =======
+
+    tuple
+        A pair (systems, condition) where:
+        - systems is a list of lists of expressions, where each sublist when solved gives
+          one component of the solution
+        - condition is a Boolean expression that must be satisfied for the system to be solvable
```

Comment:
There should be a doctest example showing a case in which the condition is not just True.

Reply:
Do you have such an example in mind ? 
I have thought about it before too but the way how we have structured the code here, it apparently keeps returning True. Maybe* because we are here trying to express all solvability conditions through the structure of the returned systems rather than through the condition parameter, all the meaningful conditions are being encoded in the structure of the solution systems themselves.

Reply:
There are examples in the tests. It can only happen when there is a symbolic parameter though.

Reply:
It should be like this but it does not work in the PR currently:
```python
In [6]: factor_poly_system([x**2 - 2], [y])
...
TypeError: expecting bool or Boolean, not `Poly(x**2 - 2, y, domain='ZZ[x]')`.
```
An example like this should be added in the tests.